### DISCUSSION TITLE
storing WorkFlowRules (Workflows) & Rules in a db

### DISCUSSION THREAD

**Discussion by asulwer** [microsoft/RulesEngine#205] [Q&A] [answered]
anyone using EF and RulesEngine models directly?  my Workflows are stored as json in my db but what about storing the models directly in the db and interfacing with DbSet?

sorry about terminology here

**Comment by tisdalestar6git** [comment]
yes  this is the table being used:
CREATE TABLE [dbo].[process_rule](
	[process_rule_id] [bigint] IDENTITY(1,1) NOT NULL,
	[execution_name] [varchar](250) NOT NULL,
	[rule_name] [varchar](250) NOT NULL,
	[create_datetime] [datetime2](7) NOT NULL,
	[success_event_num] [int] NOT NULL,
	[error_message] [varchar](250) NULL,
	[error_type] [varchar](250) NULL,
	[rule_expression] [varchar](8000) NULL,
	[rule_expression_type] [varchar](250) NULL,
 CONSTRAINT [process_rule_id_pkuc_idx] PRIMARY KEY CLUSTERED 
(
	[process_rule_id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
) ON [PRIMARY]

**Reply by asulwer** [reply]
what i am asking is something a little different.  

[RulesEngine Models](https://github.com/microsoft/RulesEngine/tree/main/src/RulesEngine/Models)

i want to add all of these, codefirst, into my db.  then use DbSet to access

**Reply by asulwer** [reply]
the models are already setup, i just to need to migrate them to my db.  why cant i?

**Comment by abbasc52** [comment, marked as answer]
@asulwer We have a new demo app with EF integration - https://github.com/microsoft/RulesEngine/tree/main/demo/DemoApp.EFDataExample

You can check it out and see if it answers your question.

**Reply by asulwer** [reply]
this is exactly what i had in mind
